coins/monero/src/wallet/scan.rs

```diff
@@ -82,11 +82,11 @@ impl Metadata {
     res.extend(self.subaddress.0.to_le_bytes());
     res.extend(self.subaddress.1.to_le_bytes());
     res.extend(self.payment_id);
-    if let Some(data) = self.arbitrary_data.as_ref() {
-      res.extend([1, u8::try_from(data.len()).unwrap()]);
-      res.extend(data);
-    } else {
-      res.extend([0]);
+
+    res.extend([u8::try_from(self.arbitrary_data.len()).unwrap()]);
```

Comment:
This check isn't correct. We can have >255 segments of data. It's just that each segment must be less than 256 in length. Please use a u32. u16 has some theoretical issues I'd rather not leave open.